src/main/scala/fft/Config.scala

```diff
@@ -20,87 +20,14 @@ import org.scalatest.Tag
 import dsptools.numbers.{Field=>_,_}
 import dsptools.numbers.implicits._
 
-import scala.collection.mutable.Map
-
-object FFTConfigBuilder {
-  def apply[T <: Data : Real](
-    id: String, fftConfig: FFTConfig, genIn: () => T, genOut: Option[() => T] = None): Config = new Config(
-    (pname, site, here) => pname match {
-      case FFTKey(_id) if _id == id => fftConfig
-      case IPXactParameters(_id) if _id == id => {
-        val parameterMap = Map[String, String]()
-
-        // Conjure up some IPXACT synthsized parameters.
-        val gk = site(GenKey(id))
-        val fftsize = fftConfig.n
-        // double these because genIn is underlying type, but input is complex
-        val totalWidthIn = genIn().getWidth 
-        val totalWidthOut = genOut.getOrElse(genIn)().getWidth
-        parameterMap ++= List(
-          ("nBands", (fftsize/gk.lanesIn).toString),
-          ("InputLanes", gk.lanesIn.toString),
-          ("InputTotalBits", totalWidthIn.toString),
-          ("OutputLanes", gk.lanesOut.toString),
-          ("OutputTotalBits", totalWidthOut.toString),
-          ("OutputPartialBitReversed", "1")
-        )
-
-        // add fractional bits if it's fixed point
-        genIn() match {
-          case fp: FixedPoint =>
-            val fractionalBits = fp.binaryPoint
-            parameterMap ++= List(
-              ("InputFractionalBits", fractionalBits.get.toString)
-            )
-          case _ =>
-        }
-        genOut.getOrElse(genIn)() match {
-          case fp: FixedPoint =>
-            val fractionalBits = fp.binaryPoint
-            parameterMap ++= List(
-              ("OutputFractionalBits", fractionalBits.get.toString)
-            )
-          case _ =>
-        }
-
-        // tech stuff, TODO
-        parameterMap ++= List(("ClockRate", "100"), ("Technology", "TSMC16nm"))
+import freechips.rocketchip.config._
 
-        parameterMap
-      }
-      case _ => throw new CDEMatchError
-    }) ++
-  ConfigBuilder.genParams(id, fftConfig.lanes, () => DspComplex(genIn(), genIn()), genOutFunc = Some(() => DspComplex(genOut.getOrElse(genIn)(), genOut.getOrElse(genIn)())))
-  def standalone[T <: Data : Real](id: String, fftConfig: FFTConfig, genIn: () => T, genOut: Option[() => T] = None): Config =
-    apply(id, fftConfig, genIn, genOut) ++
-    ConfigBuilder.buildDSP(id, {implicit p: Parameters => new FFTBlock[T]})
-}
-
-class DefaultStandaloneRealFFTConfig extends Config(FFTConfigBuilder.standalone("fft", FFTConfig(), () => DspReal()))
-class DefaultStandaloneFixedPointFFTConfig extends Config(FFTConfigBuilder.standalone("fft", FFTConfig(), () => FixedPoint(16.W, 8.BP)))
-
-class CustomStandaloneFFTConfig extends Config(FFTConfigBuilder.standalone(
-  id = "fft", 
-  fftConfig = FFTConfig(
-    n = 128,
-    lanes = 4,
-    pipelineDepth = 7
-  ), 
-  genIn = () => FixedPoint(12.W, 17.BP), 
-  genOut = Some(() => FixedPoint(15.W, 14.BP))
-))
-
-case class FFTKey(id: String) extends Field[FFTConfig]
```

Comment:
So does all this remove the IP-XACT generation? I'm okay with that, but I think future support for IP-XACT would be beneficial.

Reply:
Yeah, I haven't ported all that forward. It's not a big priority for me atm, but I agree it would be good to have.